src/main/java/org/openstreetmap/atlas/geography/boundary/CountryBoundaryMap.java

```diff
@@ -160,6 +158,87 @@
     private transient GeometryPrecisionReducer reducer;
     private final CountryListTwoWayStringConverter countryListConverter = new CountryListTwoWayStringConverter();
 
+    /**
+     * Collects leaf nodes of given {@link AbstractNode} into the given {@link MultiMap} from
+     * {@link Geometry} to {@link Envelope}s.
+     *
+     * @param root
+     *            Root node for collection
+     * @param cells
+     *            {@link MultiMap} to save cells into
+     */
+    @SuppressWarnings("unchecked")
+    static void collectCells(final AbstractNode root, final MultiMap<Geometry, Envelope> cells)
+    {
+        if (root.getLevel() > 0)
```

Comment:
If root level is always a whole number then how about
```
int rootLevel = root.getLevel();
if (rootLevel == 0)
{
    root.getChildBoundables().stream().forEach(item ->
            {
                final ItemBoundable boundable = (ItemBoundable) item;
                final Geometry polygon = (Geometry) boundable.getItem();
                final Envelope bounds = (Envelope) boundable.getBounds();
                cells.add(polygon, bounds);
            });
}
else
{
    root.getChildBoundables().stream().forEach(node ->
            {
                collectCells((AbstractNode) node, cells);
            });
}
```

I feel having an if, else would be much cleaner than if else if. What do you think?

Reply:
Theoretically root level should not be negative. Still If/else block right now is defensive. So I'd prefer to leave as-is.

Reply:
can we possibly change ```root``` -> ```inputNode```?  Root typically means level = 0 (or in the STRTree, max level).  Sometimes collectCell can take in a root node, but most times, it'll be a non-root node.